Source/player.h

```diff
@@ -147,6 +147,19 @@ enum action_id : int8_t {
 	// clang-format on
 };
 
+enum spell_flag : uint8_t {
+	// clang-format off
+	SF_ETHEREALIZE      = 1 << 0,
+	SF_RAGE_ON          = 1 << 1,
+	SF_RAGE_COOLDOWN    = 1 << 2,
+	SF_NOTHING3         = 1 << 3,
+	SF_NOTHING4         = 1 << 4,
+	SF_NOTHING5         = 1 << 5,
+	SF_NOTHING6         = 1 << 6,
+	SF_NOTHING7         = 1 << 7,
+	// clang-format on
+};

```

Comment:
New enums should be scoped (use `enum class` in this case) and use TitleCase for the enum and values (also don't need to define every value in an enum when they're unused, a comment is fine).
```suggestion
enum class SpellFlag : uint8_t {
	// clang-format off
	Etherealize  = 1 << 0,
	RageActive   = 1 << 1,
	RageCooldown = 1 << 2,
	// bits 3-7 are unused
	// clang-format on
};
```

Uses of the values will need to be updated to `SpellFlag::Etherealize` instead of `SF_ETHEREALIZE`.

`Player::_pSpellFlags` should also use this enum as the backing type, then you can use the `use_enum_as_flags` macro to make `HasAnyOf`/`HasNoneOf` (from utils/enum_traits.h) available so tests can be expressed as `if(HasAnyOf(target._pSpellFlags, SpellFlag::Etherealize)) { ... }`

Reply:
I'll update accordingly, thanks!

Reply:
@ephphatha do you still set and clear the flags the same way, or is there a different way to do that?

Reply:
Can still use bitwise operators to set and clear flags, there are template specialisations defined for `&`, `|`, `~` that become available when using the `use_enum_as_flags` macro.